mm2src/coins/eth.rs

```diff
@@ -6179,3 +6208,312 @@ async fn get_eth_gas_details(
         },
     }
 }
+
+impl ToBytes for Signature {
+    fn to_bytes(&self) -> Vec<u8> { self.to_vec() }
+}
+
+impl ToBytes for SignedEthTx {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut stream = RlpStream::new();
+        self.rlp_append(&mut stream);
+        // there is minimal but risk that stream.out() may panic
+        debug_assert!(stream.is_finished(), "RlpStream must be finished before calling out");
+        Vec::from(stream.out())
```

Comment:
Maybe use an empty vec if stream is not finished, this way we will avoid panicking and the preimage will be validated anyway and swap will fail if there is any problems in serialization.
P.S. Panicking here can be an attack vector if we are not careful since the preimage is received from the other side through a p2p message and can be made to crash the other side's node, right now it's won't be a problem probably since we use `parse_preimage` first but it can be a problem if we are not careful with code changes.

Reply:
> Maybe use an empty vec if stream is not finished

Oh, exactly! yep we can do `if stream.is_finished()` check

> right now it's won't be a problem probably since we use parse_preimage first but it can be a problem if we are not careful with code changes.

oh I see, we can refactor `trait ToBytes` impl and return `Result<Vec<u8>, Error>. Well, it will trigger other code parts, in some other traits we also have to do the refactoring and return Result type.

For now I suggest to leave `stream.is_finished()` check,
```rust
impl ToBytes for SignedEthTx {
    fn to_bytes(&self) -> Vec<u8> {
        let mut stream = RlpStream::new();
        self.rlp_append(&mut stream);
        // Handle potential panicking.
        if stream.is_finished() {
            Vec::from(stream.out())
        } else {
            // TODO: Consider returning Result<Vec<u8>, Error> in future refactoring for better error handling.
            warn!("RlpStream was not finished; returning an empty Vec as a fail-safe.");
            vec![]
        }
    }
}
```

and in next pr(s) do the refactor and return Result where we will have to. Or at least try to refactor and see, do we actually have problems with it or not. I think its better to propagate errors instead of being careful (I mean we all are humans and there is a risk of miss something).